### ISSUE TITLE
Implement first iteration of typography components

### ISSUE THREAD

**Issue by trybern** [FHIDev/Fhi.Designsystem#289]
### Acceptance criteria

- It should be implemented in accordance [with the designs in Figma](https://www.figma.com/design/VgQG6XeEbFOmHIrxyAOACR/FHI-Designsystem?node-id=5325-216213&t=Wesml3Njf7knOEiv-11)
- It should be accessable from the NPM package and CDN
- It should have automated unit and accessibility tests with appropriate coverage
- It should have a story with an introduction, usage instructions, implementation examples and action interactivity
- It should conform to the Web standard on inputs/outputs (e.g form interactions)
- It should have appropriate component level design tokens that references semantic, and if appropriate, primitive level tokens.

### Beskrivelse

FHI Designsystem har følgende tekststiler til rådighet, som skal tilbys som komponenter til bruk og nytte:

- **Display:** De største, mest iøynefallende tekststilene vi har, som passer godt til forsider, landingssider og «Hero»-seksjoner.
- **Headline:** Typiske mellomstore, overskrifter for seksjoner, artikler, kort og slikt.
- **Title:** Stiler for under- og korttitler. Typisk bruk er internt i komponenter.
- **Body:** Brødtekst, avsnitt og beskrivelser. Utgangspunkt for all tekst.
- **Label:** Etiketter på knapper, handlinger, felt og mer.

De vil – i Figma – fortsatt kun tilbys som Text Styles.

### Forslag til attributter/props
(Default i **fet skrift**.)

- Alle komponenter får `color` og `size`
- Display, Headline og Title får i tillegg `level`

#### `size`
`"large"`, **`"medium"`** eller `"small"`

Setter variant av tekststilen, og alt tilhørende (font-size, letter-spacing, line-height og font-weight).

(Her tenker jeg også vi må bestemme oss for om de skal ha default margin-bottom (og -top) eller ikke.)

#### `color`
**`(neutral-)text-default`**

Setter farge på teksten.

> [!NOTE]
>  **Nice-to-have:** Her bør det gis mulighet til å skrive så lite av navnet på variabelen som mulig, for eksempel «subtle» og «inverted» (for å skifte inni *Neutral*-skalaen). Dette kan vi potensielt se på i fremtiden.

> [!NOTE]
>  En del designsystemer har en metode for å la farge «dryppe» ned i komponenter som ligger inne i en annen. For eksempel kan man sette `"danger"` på en `<dialog>`, hvor tekst, knapper og mer også får dette som utgangspunkt. Dette er noe vi kan utforske i fremtiden. 

#### `level`
`"1"`, `"2"`, `"3"`, `"4"`, `"5"` eller `"6"`.

Setter nivå i strukturen (`level="3"` blir `h3`). Gjelder kun Display, Headline og Title. 

(I første utkast var tanken å ha `as` på alle elementer )

<details><summary><strong>Til senere?</strong></summary>
Vi holder det slankt og greit i første runde. Her er noen umiddelbare notater som legger igjen her til senere runder, uten at det betyr at vi kommer til å inkludere det på et fremtidig tidspunkt.

#### `as` (eller `tag`?)
**`"p"`** (Body) og **`"label"`** (Label)

Overstyrer semantisk HTML-element (f.eks. `as="legend"` viser det som et `<legend>`-element). Gjelder kun Body og Label.

Det finnes situasjoner hvor man vil overstyre overskriftene med noe annet enn `h1` (osv.), for eksempel ingresser og slikt med samme stil, men som egentlig bør være en `p`. Her tenker jeg vi holder litt tilbake for å venne folk til `level`, men samtidig som vi samler behov og brukssituasjoner for `as` på Display, Headline og Title.

#### `no-margin`
Setter marginer til 0 slik at brukerne selv setter dette, for eksempel med gap og whatnot. Dette forutsetter at vi ender opp med å servere standardmarginer til størrelsene.

#### `long`
Gir Body en linjehøyde som passer lengre avsnitt bedre, for å gjøre det lettere å lese.

#### `weight`
Setter font-weight. For eksempel har Radio labels i Regular, mens default er Medium. Vi vil også at Bold skal være vår token, som er dyttet ned til Semibold. Enn så lenge kan vi nok håndtere dette internt i komponentene med `<strong>`/`<b>`.

#### `truncate`
Alternativt `max-lines`, `clamp` eller lignende. Man kan sette en maksgrense på antall linjer tekst vises. For eksempel kan man avkorte et avsnitt som overstiger 2 linjer, med ellipser (…) på slutten.

Noen løser dette med egne «utility»-komponenter (f.eks. `<Truncate>`) med tekstkomponenter inni igjen. 

#### `<validation-message>`
Eget komponent for tilbakemeldinger, for eksempel med rødt kryss ved feil eller grønn hake ved godkjent validering?
</details>
